### PR TITLE
feat(catalog): Remove single OneOf from AnyOf

### PR DESCRIPTION
**Pull request by lordrip**
Extract single OneOf definition from AnyOf definition and put it into the root definitions.

relates: https://github.com/KaotoIO/kaoto-next/issues/948
relates: https://github.com/KaotoIO/kaoto-next/issues/560

### CODE REVIEW

File: packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/test/java/io/kaoto/camelcatalog/CamelYamlDslSchemaProcessorTest.java [hunk 1]
```diff
@@ -51,6 +51,18 @@ public void testProcessSubSchema() throws Exception {
         assertEquals("object", beanPropertySchema.get("type").asText());
     }
 
+    @Test
+    public void test_extract_single_oneOf_from_anyOf() throws Exception {
```
Comment (igarashitm) [inline packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/test/java/io/kaoto/camelcatalog/CamelYamlDslSchemaProcessorTest.java:55]: maybe `testExtractSingleOneOfFromAnyOf()` - we rarely use C style underscore in Java

**Review by igarashitm — APPROVED** [review]
Commented about the test method naming, LGTM otherwise

**Review by shivamG640 — APPROVED** [review]
I can see the changes working for  the `camelYamlDsl-errorHandler` Json.

File: packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/test/java/io/kaoto/camelcatalog/CamelYamlDslSchemaProcessorTest.java (hunk 1 above)
Comment (lordrip) [inline packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/test/java/io/kaoto/camelcatalog/CamelYamlDslSchemaProcessorTest.java:55]:
```suggestion
    public void testExtractSingleOneOfFromAnyOf() throws Exception {
```

**Pull request merged**